src/peripherals.rs

```diff
@@ -0,0 +1,182 @@
+use std::error::Error;
+
+pub enum Peripherals {
+    Ads1115, // ADC
+    Ak09915, // 3-axis magnetometer
+    Bmp280,  // Pressure and Temperature
+    Gpio,
+    Icm20689, // Accelerometer and Gyroscope
+    Leak,
+    Pca9685, // PWM controller
+    Sk6812,  // Neopixel RGB LED
+}
+
+pub enum PeripheralClass {
+    Accelerometer,
+    Adc,
+    DigitalInput,
+    DigitalOutput,
+    Gyroscope,
+    Led,
+    Magnetometer,
+    Pressure,
+    Pwm,
+    RgbLed,
+    Temperature,
+}
+
+pub struct PeripheralInfo {
+    pub peripheral: Peripherals,
+    pub class: Vec<PeripheralClass>,
+}
+
+pub trait AnyHardware {
+    fn as_any(&self) -> &dyn std::any::Any;
+    fn as_any_mut(&mut self) -> &mut dyn std::any::Any;
+
```

Comment:
```suggestion
pub trait AnyHardwareExt {
    fn as_any(&self) -> &dyn std::any::Any;
    fn as_any_mut(&mut self) -> &mut dyn std::any::Any;
}

impl<T: AnyHardware + 'static> AnyHardwareExt for T {
    fn as_any(&self) -> &dyn std::any::Any {
        self
    }

    fn as_any_mut(&mut self) -> &mut dyn std::any::Any {
        self
    }
}

pub trait AnyHardware {

```

we can auto-implement that, so we remove those ten lines for each sensor :)